control-plane/connect-inject/webhook/consul_dataplane_sidecar_test.go

```diff
@@ -1299,6 +1300,148 @@ func TestHandlerConsulDataplaneSidecar_Metrics(t *testing.T) {
 	}
 }
 
+func TestHandlerConsulDataplaneSidecar_Lifecycle(t *testing.T) {
```

Comment:
Can we add a multiport test case here?

Reply:
We already take the proxy lifecycle into account in the multiport test. It fails without the graceful flags set. I think the lifecycle test would pretty much be a duplicate of the Multiport test?

https://github.com/hashicorp/consul-k8s/blob/acceptance/proxy-lifecycle-shutdown-fixed/control-plane/connect-inject/webhook/consul_dataplane_sidecar_test.go#L626

Reply:
yep you're right, that's exactly the case I was thinking about! I wanted to make sure the graceful port also got incremented